src/cpp/common/writer.h

```diff
@@ -97,5 +108,32 @@ class writer
     return m_metadata[key];
   }
 };
+
+class config_writer: public writer
+{
+  std::map<std::string, std::map<std::string, std::vector<std::shared_ptr<writer>>>> m_output;
+  partition_info m_partition;
+
+public:
+  const std::map<std::string, std::map<std::string,  std::vector<std::shared_ptr<writer>>>>&
+  get_kernel_map() const { return m_output; }
+
+  void add_kernel_map(const std::string& kernel, const std::string& instance, std::vector<std::shared_ptr<writer>> val) {
+    m_output[kernel][instance] = std::move(val);
+  }
+
+  uint32_t get_numcore() const { return m_partition.get_numcore(); }
+
+  uint32_t get_numcolumn() const { return m_partition.get_numcolumn(); }
+
+  uint32_t get_nummem() const { return m_partition.get_nummem(); }
+
+  void set_numcolumn(uint32_t val) { m_partition.set_numcolumn(val); }
```

Comment:
Once set what prevents some one to call the same API again and override the values? Shouldn't the partition info be passed in in the constructor of this class so it cannot be changed later?